document-parsers/langchain4j-document-parser-apache-tika/src/main/java/dev/langchain4j/data/document/parser/apache/tika/ApacheTikaDocumentParser.java

```diff
@@ -76,19 +79,21 @@ public ApacheTikaDocumentParser(Parser parser,
      * @param contentHandlerSupplier Supplier for Tika content handler. Default: {@link BodyContentHandler} without write limit
      * @param metadataSupplier       Supplier for Tika metadata. Default: empty {@link Metadata}
      * @param parseContextSupplier   Supplier for Tika parse context. Default: empty {@link ParseContext}
+     * @param includeMetadata        Whether to include metadata in the parsed document
      */
-    public ApacheTikaDocumentParser(Supplier<Parser> parserSupplier,
-                                    Supplier<ContentHandler> contentHandlerSupplier,
-                                    Supplier<Metadata> metadataSupplier,
-                                    Supplier<ParseContext> parseContextSupplier) {
+    public ApacheTikaDocumentParser(
```

Comment:
This is a breaking change, please create a new ctor with 5 fields and deprecate this one

Reply:
So the only way user can include metadata is to use this ctor that forces user to configure lots of suppliers? Perhaps we should have a more user friendly way to do so? Like `new ApacheTikaDocumentParser(true)` or `ApacheTikaDocumentParser.builder().includeMetadata(true).build()`, WDYT?

Reply:
@langchain4j Here you go: https://github.com/langchain4j/langchain4j/pull/2192/commits/8c2fade93a98c9638a995cbd8c9b01038bf72925